src/components/VMTable.vue

```diff
@@ -135,11 +135,11 @@ export default {
 
     getProgramLabel (program) {
       if (!('extra_fields' in program.content)) {
-        return program.item_hash
-      } else if (!('program_name' in program.content.extra_fields)) {
-        return program.item_hash
-      } else if (program.content.extra_fields.program_name.length === 0) {
-        return program.item_hash
+        if (program.content?.metadata?.name) {
+          return `${program.content?.metadata?.name} (${ellipseAddress(program.item_hash)})`
+        } else {
+          return program.item_hash
```

Comment:
In case no name has been assigned to a program, displaying the complete hash on a screen size greater than 768px (Tablets) seems more coherent to me but I can ellipse it for a smaller screen size, what do you think?

Reply:
Please to, that would also keep the behaviour coherent.

Reply:
Done! Hash will be ellipsed depending on the size of the screen